api/src/services/workspace/table-record/table-record.hooks.ts

```diff
@@ -0,0 +1,226 @@
+        console.log(workspaceSlug, datasourceSlug, tableSlug, context.params.query)
+        context.params.$$id =
+          'WS_' +
+          (workspaceSlug as string) +
+          '_DS_' +
+          (datasourceSlug as string) +
+          '_TBL_' +
+          (tableSlug as string)
+
+        // if validator already exist, we don't need to compute the typebox schema
+        if (!validators[context.params.$$id]) {
+          const tableResponse = await context.app.service(SERVICES.WORKSPACE_TABLE).find({
+            query: {
+              slug: tableSlug,
+              $eager: '[fields,relations.[toTable]]',
+            },
+            route: {
+              workspaceSlug,
+              datasourceSlug,
+            },
+          })
+
+          if (tableResponse.total !== 1)
+            throw new NotFound(
+              'Table ' +
+                (tableSlug as string) +
+                ' not found in workspace ' +
+                (workspaceSlug as string),
+            )
+
+          const table = tableResponse.data[0]
+          const tableSchema: Record<string, TSchema> = {}
+
+          table.fields?.forEach((f) => {
+            tableSchema[f.slug] = convertLocoKitFieldTypeToTypeboxSchema(f)
+          })
+          const tableRelationsNames: string =
+            table.relations
+              ?.reduce((acc, r) => {
+                console.log(r.settings)
+                acc.push(r.settings.toTable)
+                return acc
+              }, [] as string[])
+              .join('|') ?? ''
+          const tableRelationRegexp = new RegExp(
+            `^(${tableRelationsNames})|\\[(${tableRelationsNames})(,(${tableRelationsNames})(?!.*\\2))*\\]$`,
+          )
+          console.log(tableRelationsNames, tableRelationRegexp)
+
+          const tableQuerySchema = Type.Intersect(
+            [
+              querySyntax(
+                Type.Object(tableSchema, {
+                  additionalProperties: false,
+                }),
+              ),
+              Type.Object(
+                {
+                  $joinRelated: Type.Optional(Type.RegEx(tableRelationRegexp)),
+                  $joinEager: Type.Optional(Type.RegEx(tableRelationRegexp)),
+                  $eager: Type.Optional(Type.RegEx(tableRelationRegexp)),
+                },
+                {
+                  additionalProperties: false,
+                },
+              ),
+            ],
+            {
+              $id: context.params.$$id,
+              // additionalProperties: false
+            },
+          )
+          context.params.$$schema = tableQuerySchema
+        }
+
+        await next()
+      },
+      async function computeValidator(context: HookContext, next: NextFunction) {
+        // try to find the validator in the validators Record
+        if (!validators[context.params.$$id]) {
+          // TODO: is this safe as it could grow a lot in the mid term running ?
```

Comment:
Maybe we can create an issue to reminder this possible issue?

Reply:
Yup, I'll add in the description benchmark tests too.

refer to #233